### ISSUE TITLE
[Epic] Heading consistency across Grafana (incorrect hierarchy, additional headings required, incorrect headings)

### ISSUE THREAD

**Issue by ckbedwell** [grafana/grafana#73693]
# User experience
In certain pages and areas throughout Grafana the heading hierarchy is inconsistent and jumps around and makes it difficult for users of assistive technologies to orientate themselves and navigate the page. 

Many of these issues stem from the tight coupling of semantic html headers with their design. Now the [Text component](https://github.com/grafana/grafana/blob/main/packages/grafana-ui/src/components/Text/Text.tsx) has been added we can easily specify the semantic element whilst decoupling it from its intended design.

[w3 page structure tutorial for headings.](https://www.w3.org/WAI/tutorials/page-structure/headings/)
[Deque University heading levels should only increase by one](https://dequeuniversity.com/rules/axe/4.4/heading-order)
[TPGi article discussing headings](https://www.tpgi.com/heading-off-confusion-when-do-headings-fail-wcag/)

```[tasklist]
### Incorrect hierarchy
- [ ] Welcome page / welcome dashboard heading hierarchy `/` (dashboard components)
- [ ] Some explore page datasource components use the wrong headings and create hierarchy issues _(more investigation needed)_
- [ ] Alert rules heading hierarchy`/alerting/list`
- [ ] Contact points heading hierarchy `/alerting/notifications`
- [ ] Contact points `/alerting/notifications`
- [ ] Incidents plugin is not available h3 should be h2 `a/grafana-incident-app/incidents`
- [ ] Connections -> add a new connection Data sources should be h2, Datasources should be h3s
```

```[tasklist]
###  Existing heading hierarchy issues
- [ ] https://github.com/grafana/grafana/issues/66554
- [ ] https://github.com/grafana/grafana/issues/66560
- [ ] https://github.com/grafana/grafana/issues/67040
```

```[tasklist]
### Additional headings required
- [ ] Dashboards page folders should be h2s and dashboards should be h3s `/dashboards`
- [ ] Explore page lacks a h1, queries should be h2s `/explore`
- [ ] Some explore page datasource components use no headers _(more investigation needed)_
- [ ] Alerting Notification policies policies need h2s `alerting/routes`
- [ ] Alerting Groups page needs grouping h2s `/alerting/groups`
```

```[tasklist]
### Incorrectly labelled as a heading
- [ ] Welcome and Getting Started components have an h6 for 'Untitled' which is not needed.
- [ ] Machine learning holidays zero state should not be an h5
- [ ] Alerting Ui app zero states (e.g. no rules found / no alertmanager data sources configured) should not be an h5
```

```[tasklist]
### Yet to be evaluated
- [ ] k6 app
- [ ] k8s app
- [ ] Observability app
- [ ] Pyroscope app
- [ ] kowalski app
- [ ] Synthetics app
```

**Comment by tskarhed** [comment]
I have written a design doc about how heading level could be automatically handled. The hard part is getting it to play nicely with native elements - maybe the two need to work in parallel, and we explicitly use "dynamic" headings on components we know will be used in different context with different priorities.

**Comment by tskarhed** [comment]
From a frontend platform POV, I think we can find common patterns and provide guidelines and possibly better components for other teams to solve this issue.

**Comment by tskarhed** [comment]
I made a small POC to try to address some parts of this issue https://github.com/grafana/grafana/pull/97408

**Comment by taleena** [comment]
Looks like the above pull was closed due to inactivity, perhaps we should move what it fixes out into it's own issue?

**Comment by taleena** [comment]
Adding a tool to help us catch/report this in development, ideally something that isn't easily ignored.

**Comment by ashharrison90** [comment]
closing as duplicate of https://github.com/grafana/grafana/issues/131389
